### PR TITLE
Reimplement edit command

### LINKED ISSUE #108: Reimplement edit command

### PR DESCRIPTION
**Pull request by SamsonChew**
closes #108

### CODE REVIEW

File: docs/UserGuide.md [hunk 1]
```diff
@@ -225,6 +225,22 @@ section for detailed information
...
+**Examples:**
+
+- `edit 1 n/Samson Chew s/A1234567M`  
+  Edits the name and student ID of the 1st student to be "Samson Chew" and "A1234567M" respectively.
+
+- `edit 2 c/1002`  
+  Edits the tutorial ID of the 2nd student to be "1002".
+
+This command allows you to specify new values for a student's name, student ID, and/or tutorial ID, ensuring that each student's information is up to date and accurately reflects any changes in their academic or tutorial assignments.
+
```
Comment (waihongteh) [inline docs/UserGuide.md:243]: Nice and detail documented user guide!

**Review by waihongteh — APPROVED** [review]
All changes made look good to me! Feel free to merge.

File: src/test/java/seedu/address/logic/commands/EditCommandTest.java [hunk 2]
```diff
@@ -160,6 +135,61 @@ public void execute_invalidStudentIndexFilteredList_failure() {
...
+
+        EditCommand.EditStudentDescriptor descriptor = new EditStudentDescriptorBuilder()
+                .withTutorialId(nonExistentTutorialId).build();
+        EditCommand editCommand = new EditCommand(indexFirstStudent, descriptor);
+
+        String expectedMessage = EditCommand.MESSAGE_TUTORIAL_NOT_FOUND + nonExistentTutorialId;
+
+        assertCommandFailure(editCommand, model, expectedMessage);
+    }
+
```
Comment (tansanchian) [inline src/test/java/seedu/address/logic/commands/EditCommandTest.java:192]: These test cases have good coverage. Great!

**Review by tansanchian — APPROVED** [review]
LGTM!

File: src/test/java/seedu/address/logic/parser/EditCommandParserTest.java [hunk 3]
```diff
@@ -78,132 +64,119 @@ public void parse_invalidPreamble_failure() {
...
-                + TAG_DESC_FRIEND + PHONE_DESC_AMY + ADDRESS_DESC_AMY + EMAIL_DESC_AMY + TAG_DESC_FRIEND
-                + PHONE_DESC_BOB + ADDRESS_DESC_BOB + EMAIL_DESC_BOB + TAG_DESC_HUSBAND;
+        // Multiple valid fields repeated
+        userInput = targetIndex.getOneBased() + NAME_DESC_AMY + STUDENTID_DESC_AMY + TUTORIALID_DESC_AMY
+                + NAME_DESC_AMY + STUDENTID_DESC_AMY + TUTORIALID_DESC_AMY;
 
-        assertParseFailure(parser, userInput,
-                Messages.getErrorMessageForDuplicatePrefixes(PREFIX_PHONE, PREFIX_EMAIL, PREFIX_ADDRESS));
+        assertParseFailure(parser, userInput, getErrorMessageForDuplicatePrefixes(
+                PREFIX_NAME, PREFIX_STUDENTID, PREFIX_TUTORIALID));
```
Comment (tansanchian) [inline src/test/java/seedu/address/logic/parser/EditCommandParserTest.java:141]: I like how you change all the local fields to shared class fields. Good job!

**Review by tansanchian — APPROVED** [review]
Great!

**Pull request merged**